Applications/Utils/MeshGeoTools/createIntermediateRasters.cpp

```diff
@@ -0,0 +1,141 @@
+                                            "", "output.asc");
+    cmd.add(output_arg);
+    TCLAP::ValueArg<std::size_t> number_arg(
+        "n", "number", "number of rasters to be calculated", false, 1, "int");
+    cmd.add(number_arg);
+
+    cmd.parse(argc, argv);
+
+    std::unique_ptr<GeoLib::Raster> dem1(
+        FileIO::AsciiRasterInterface::readRaster(input1_arg.getValue()));
+    std::unique_ptr<GeoLib::Raster> dem2(
+        FileIO::AsciiRasterInterface::readRaster(input2_arg.getValue()));
+
+    GeoLib::RasterHeader const h1 = dem1->getHeader();
+    GeoLib::RasterHeader const h2 = dem2->getHeader();
+
+    std::size_t errors_found(0);
+    if (h1.origin[0] != h2.origin[0])
+    {
+        ERR("Origin x-coordinate is not the same in both raster files.\n");
+        errors_found++;
+    }
+    if (h1.origin[1] != h2.origin[1])
+    {
+        ERR("Origin y-coordinate is not the same in both raster files.\n");
+        errors_found++;
+    }
+    if (h1.cell_size != h2.cell_size)
+    {
+        ERR("Cellsize is not the same in both raster files.\n");
+        errors_found++;
+    }
+    if (h1.n_cols != h2.n_cols)
+    {
+        ERR("Raster width is not the same in both raster files.\n");
+        errors_found++;
+    }
+    if (h1.n_rows != h2.n_rows)
+    {
+        ERR("Raster height is not the same in both raster files.\n")
+        errors_found++;
+    }
+
+    if (errors_found > 0)
+        return 2;
+
+    std::size_t const n = number_arg.getValue();
+    std::vector<std::vector<double>> raster;
+    for (std::size_t i = 0; i < n; ++i)
+    {
+        std::vector<double> r;
+        r.reserve(h1.n_cols * h1.n_rows);
+        raster.push_back(r);
+    }
+
+    auto it2 = dem2->begin();
+    for (auto it1 = dem1->begin(); it1 != dem1->end(); ++it1)
+    {
+        if (it2 == dem2->end())
+        {
+            ERR("Error: File 2 is shorter than File 1.");
+            return 1;
+        }
+        if (*it1 == h1.no_data || *it2 == h2.no_data)
+        {
+            for (std::size_t i = 0; i < n; ++i)
+                raster[i].push_back(h1.no_data);
+        }
+        else
+        {
+            double const min = std::min(*it1, *it2);
+            double const max = std::max(*it1, *it2);
+            double const step = (max - min) / static_cast<double>(n + 1);
+            for (std::size_t i = 0; i < n; ++i)
+                raster[i].push_back(min + ((i+1) * step));
+        }
+        it2++;
+    }
+    if (it2 != dem2->end())
+    {
+        ERR("Error: File 1 is shorter than File 2.");
+        return 1;
+    }
+
+    std::string filename = output_arg.getValue();
+    for (std::size_t i = 0; i < n; ++i)
+    {
+        std::string const basename = BaseLib::dropFileExtension(filename);
+        std::string const ext = BaseLib::getFileExtension(filename);
+
+        GeoLib::RasterHeader h (h1);
+        GeoLib::Raster r(std::move(h), raster[i].begin(), raster[i].end());
```

Comment:
Why not simply make a copy when passing `h1` to Raster ctor?
```suggestion
        GeoLib::Raster r(h1, raster[i].begin(), raster[i].end());
```

Reply:
:heavy_check_mark: